content/en/docs/features/control-plane/entity-tag.md

```diff
@@ -0,0 +1,217 @@
+
+```c
+int ret = oc_main_init(&handler);
+...
+// Create all custom resources
+...
+oc_etag_load_and_clear();
+while (!quit)
+{
+    oc_clock_time_t next_event_mt = oc_main_poll_v1();
+    ...
+}
+oc_etag_dump();
+...
+// Remove all custom resources
+...
+oc_main_shutdown();
+```
+
+### Edge Cases
+
+1. **Time Synchronization**: Maintaining accurate time synchronization is crucial for generating correct epoch time used in ETAGs. Inconsistent system clock synchronization between the device and client can lead to ETAG conflicts. To address this, if the epoch time is less than the global ETAG, the global ETAG is used as a mitigation strategy. A random number is added to the global ETAG as described in the [ETAG Generation Algorithm](#etag-generation-algorithm) to ensure uniqueness.
+2. **Storage Failure**: In the event of ETAG storage failure, the loss of ETAG information can result in inconsistencies when determining resource modification. To mitigate this issue, the device developer should carefully check the return value from the `oc_etag_dump` function to confirm the success of the operation. Storage failures often occur due to insufficient storage space.
+3. **ETAG Reset**: When a system restarts without a proper dump or due to a power outage, the global ETAG stored in memory is lost. To mitigate this, the ETAGs are not loaded through the `oc_etag_load_and_clear` function, which requires the client to resynchronize with the device to establish a new global ETAG.
+4. **Absence of Resources after Device Reboot**: ETAGs for non-existing resources are taken into account when calculating the global ETAG during the `oc_etag_load_and_clear` process. This ensures that ETAGs assigned to previous resources are not reused by new resources, thereby avoiding conflicts and maintaining consistency.
+5. **Addition of Resources after Device Reboot**: When new resources are added after a device reboot, the ETAG is set using the algorithm described in the [ETAG Generation Algorithm](#etag-generation-algorithm). This ensures proper tracking of changes for the newly added resource and maintains synchronization with the global ETAG.
+6. **Global ETAG Overflow**: In the event that the global ETAG reaches its maximum value and overflows, all resource ETAGs must be reinitialized. This prevents potential conflicts and ensures the continued accurate tracking of resource changes.
+
+## Efficient Device Twin Synchronization
```

Comment:
When the user tries this feature, and he doesn't see any performance optimization, how can he quickly identify which resource/s were changed during the reboot or the time it was offline?

Reply:
added tip